src/aqua/decider.aqua

```diff
@@ -189,90 +190,122 @@ func join_deal(spell_id: string, block_number: string, deal: DealCreatedData) ->
             on worker_id:
                 args = WorkerArgs(deal_id = deal.deal_id, worker_def_cid = deal.app_cid, ipfs = settings!.worker_ipfs)
                 worker_spell_id <- PeerSpell.install(settings!.worker_script, args, settings!.worker_config)
-                log(spell_id, ["created worker for deal", deal.deal_id, "spell_id", worker_spell_id, "worker_id", worker_id])
+                log(["created worker for deal", deal.deal_id, "spell_id", worker_spell_id, "worker_id", worker_id])
                 joined_deal = JoinedDeal(deal_id = deal.deal_id, spell_id = worker_spell_id, worker_id = worker_id)
                 store_joined_deal(spell_id, joined_deal)
-                deal_state = DealState(from_block = block_number, app_cid = deal.app_cid)
+                deal_state = DealState(from_block = block_number)
                 store_deal_state(spell_id, deal.deal_id, deal_state)
                 status <<- true
 
         catch e:
-            log(spell_id, ["cannot create worker", deal.deal_id, e.message, "; skip"])
+            log(["cannot create worker", deal.deal_id, e.message, "; skip"])
             status <<- false
 
     <- status!
 
 func process_deal(spell_id: string, deal: DealCreated):
+    log = (msg: []⊤):
+            spell_log(spell_id, msg)
+
     if decide(deal.info) == false:
-        log(spell_id, ["skipping deal for deal id", deal.info.deal_id, "from_block", deal.block_number])
+        log(["skipping deal for deal id", deal.info.deal_id, "from_block", deal.block_number])
     else:
-        log(spell_id, ["joining the deal", deal.info.deal_id, "from_block", deal.block_number])
+        log(["joining the deal", deal.info.deal_id, "from_block", deal.block_number])
         is_ok <- join_deal(spell_id, deal.block_number, deal.info)
         if is_ok:
-            log(spell_id, ["joined the deal", deal.info.deal_id])
+            log(["joined the deal", deal.info.deal_id])
         else:
-            log(spell_id, ["couldn't join the deal", deal.info.deal_id])
+            log(["couldn't join the deal", deal.info.deal_id])
+
+func is_less(listener_id: string, latest_block: string, to_block: string) -> bool:
+    FluenceAuroraConnector listener_id
+    is: *bool
+
+    -- if to_block is less then latest block, move from_block to to_block
```

Comment:
Seems like this comment is misplaced. It should be around `need_update <- is_less`